exchanges/poloniex/poloniex_test.go

```diff
@@ -142,900 +84,2645 @@ func TestGetFee(t *testing.T) {
+		{Event: "subscribe", Channel: []string{"tickers"}, Symbols: enabledPairs.Strings()},
+		{Event: "subscribe", Channel: []string{"book_lv2"}, Symbols: enabledPairs.Strings()},
+	}
+
+	result := e.handleFuturesSubscriptions("subscribe", subscs)
+	require.Len(t, payloads, 2)
+	for i := range subscs {
+		require.Equal(t, payloads[i], result[i])
 	}
 }
 
-func TestGetTimestamp(t *testing.T) {
+func TestWebsocketSubmitOrders(t *testing.T) {
 	t.Parallel()
-	st, err := e.GetTimestamp(t.Context())
-	require.NoError(t, err)
+	_, err := e.WebsocketSubmitOrders(t.Context(), nil)
+	require.ErrorIs(t, err, common.ErrFunctionNotSupported)
+}
 
-	if st.IsZero() {
-		t.Error("expected a time")
-	}
+func TestWebsocketModifyOrder(t *testing.T) {
+	t.Parallel()
+	_, err := e.WebsocketModifyOrder(t.Context(), nil)
+	require.ErrorIs(t, err, common.ErrFunctionNotSupported)
 }
 
-func TestGetServerTime(t *testing.T) {
+func TestOrderbookLevelFromSlice(t *testing.T) {
 	t.Parallel()
-	st, err := e.GetServerTime(t.Context(), asset.Spot)
+	var obData []types.Number
+	data := []byte(`["88350.22","0.019937","88376.19","0.000203","88376.58","0.000696"]`)
+	err := json.Unmarshal(data, &obData)
 	require.NoError(t, err)
 
-	if st.IsZero() {
-		t.Error("expected a time")
+	target := []orderbook.Level{{Price: 88350.22, Amount: 0.019937}, {Price: 88376.19, Amount: 0.000203}, {Price: 88376.58, Amount: 0.000696}}
+	obLevels := orderbookLevelFromSlice(obData)
+	require.Len(t, obLevels, len(target))
+	for x := range obLevels {
+		require.Equal(t, target[x].Price, obLevels[x].Price)
+		require.Equal(t, target[x].Amount, obLevels[x].Amount)
 	}
 }
 
-func TestFetchTradablePairs(t *testing.T) {
+var channelIntervals = []struct {
+	input    string
+	channel  string
+	interval kline.Interval
+	err      error
+}{
+	{input: "mark_candles_hour_1", channel: "mark_candles", interval: kline.OneHour},
+	{input: "mark_price_candles_minute_1", channel: "mark_price_candles", interval: kline.OneMin},
+	{input: "mark_candles_minute_30", channel: "mark_candles", interval: kline.ThirtyMin},
+	{input: "index_candles_hour_4", channel: "index_candles", interval: kline.FourHour},
+	{input: "candles_minute_30", channel: "candles", interval: kline.ThirtyMin},
+	{input: "mark_candles_day_3", channel: "mark_candles", interval: kline.ThreeDay},
+	{input: "mark_candles_hour_abc", channel: "mark_candles", interval: kline.Interval(0), err: kline.ErrUnsupportedInterval},
+}
+
+func TestChannelToIntervalSplit(t *testing.T) {
 	t.Parallel()
-	_, err := e.FetchTradablePairs(t.Context(), asset.Spot)
-	if err != nil {
-		t.Error(err)
+	for _, chd := range channelIntervals {
+		t.Run(chd.input, func(t *testing.T) {
+			t.Parallel()
+			c, i, err := channelToIntervalSplit(chd.input)
+			require.ErrorIs(t, err, chd.err)
+			require.Equal(t, chd.channel, c)
+			assert.Equal(t, chd.interval, i)
+		})
 	}
 }
 
-func TestGetCurrencyTradeURL(t *testing.T) {
+func TestStatusCodeUnmarshal(t *testing.T) {
```

Comment:
🚧 Change request:
* There's no such type as StatusCode
* We're testing `Error` not `Unmarshal`
```suggestion
func TestStatusResponseError(t *testing.T) {
```